remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/verification/PublicKeyCache.kt

```diff
@@ -1,22 +1,38 @@
 package com.rakuten.tech.mobile.remoteconfig.verification
 
 import android.content.Context
+import android.os.Build
 import androidx.annotation.VisibleForTesting
 import com.rakuten.tech.mobile.remoteconfig.api.PublicKeyFetcher
 
 internal class PublicKeyCache @VisibleForTesting constructor(
     private val keyFetcher: PublicKeyFetcher,
-    context: Context
+    context: Context,
+    private val encryptor: Encryptor
 ) {
 
+    constructor(
+        keyFetcher: PublicKeyFetcher,
+        context: Context
+    ) : this(
+        keyFetcher,
+        context,
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+            AesEncryptor()
+        else
+            RsaEncryptor(context)
```

Comment:
What happens if a key encrypted with RSA is decrypted with AES? (e.g. the user upgraded their OS).
I guess it's ok cos it will fetch and cache a new key when the config is fetched?

Reply:
That's a good point, it would throw an exception, but then it wouldn't generate a new key. I've added a fix for this so it will generate a new key if the stored on is the wrong type.

It would also be possible to use RSA encryption on all platforms and simplify things a bit. But I've read that AES encryption is much faster and seems to be the preferred method, so I'd like to use the better method where possible.